test/powershell/Modules/Microsoft.PowerShell.Management/Set-Content.Tests.ps1

```diff
@@ -1,51 +1,102 @@
 Describe "Set-Content cmdlet tests" -Tags "CI" {
-  $file1 = "file1.txt"
-  Setup -File "$file1" -Content $file1
-  Context "Set-Content should actually set content" {
-    It "should set-Content of testdrive:\$file1" {
-      $result=set-content -path testdrive:\$file1 -value "ExpectedContent" -passthru 
-      $result| Should be "ExpectedContent"
+    BeforeAll {
+        $file1 = "file1.txt"
+        $filePath1 = join-path $testdrive $file1
+        # if the registry doesn't exist, don't run those tests
+        $skipRegistry = ! (test-path hklm:/)
+        #if not Admin, don't run these tests
+        if($IsWindows)
+        {
+            $windowsIdentity = [Security.Principal.WindowsIdentity]::GetCurrent()
+            $windowsPrincipal = new-object 'Security.Principal.WindowsPrincipal' $windowsIdentity
+            $SkipNotAdmin = $windowsPrincipal.IsInRole("Administrators") -eq $false
```

Comment:
>$SkipNotAdmin = $windowsPrincipal.IsInRole("Administrators") -eq $false [](start = 12, length = 71)

we have a new allowed tag: `RequireAdminOnWindows` which is applicable to a `Describe` block
the test below can be moved to it's own describe block and tagged with `RequireAdminOnWindows`
(see New-Item.Tests.ps1)